src/connector/jdbc/src/main/java/com/taosdata/jdbc/ws/JSONResultSet.java

```diff
@@ -0,0 +1,593 @@
+            case TSDBConstants.TSDB_DATA_TYPE_FLOAT:
+                return row.getFloat(colIndex);
+            case TSDBConstants.TSDB_DATA_TYPE_DOUBLE:
+                return row.getDouble(colIndex);
+            case TSDBConstants.TSDB_DATA_TYPE_TIMESTAMP:
+                return parseTimestampColumnData(row, colIndex);
+            case TSDBConstants.TSDB_DATA_TYPE_BINARY:
+                return row.getString(colIndex) == null ? null : row.getString(colIndex).getBytes();
+            case TSDBConstants.TSDB_DATA_TYPE_NCHAR:
+                return row.getString(colIndex) == null ? null : row.getString(colIndex);
+            case TSDBConstants.TSDB_DATA_TYPE_JSON:
+                //  all json tag or just a json tag value
+                return row.get(colIndex) != null && (row.get(colIndex) instanceof String || row.get(colIndex) instanceof JSONObject)
+                        ? JSON.toJSONString(row.get(colIndex), SerializerFeature.WriteMapNullValue)
+                        : row.get(colIndex);
+            default:
+                return row.get(colIndex);
+        }
+    }
+
+    private Timestamp parseTimestampColumnData(JSONArray row, int colIndex) throws SQLException {
+        if (row.get(colIndex) == null)
+            return null;
+        String tsFormatUpperCase = this.statement.getConnection().getClientInfo(TSDBDriver.PROPERTY_KEY_TIMESTAMP_FORMAT).toUpperCase();
+        TimestampFormat timestampFormat = TimestampFormat.valueOf(tsFormatUpperCase);
+        switch (timestampFormat) {
+            case TIMESTAMP: {
+                Long value = row.getLong(colIndex);
+                //TODO: this implementation has bug if the timestamp bigger than 9999_9999_9999_9
+                if (value < 1_0000_0000_0000_0L) {
+                    this.timestampPrecision = TimestampPrecision.MS;
+                    return new Timestamp(value);
+                }
+                if (value >= 1_0000_0000_0000_0L && value < 1_000_000_000_000_000_0l) {
+                    this.timestampPrecision = TimestampPrecision.US;
+                    long epochSec = value / 1000_000L;
+                    long nanoAdjustment = value % 1000_000L * 1000L;
+                    return Timestamp.from(Instant.ofEpochSecond(epochSec, nanoAdjustment));
+                }
+                if (value >= 1_000_000_000_000_000_0l) {
+                    this.timestampPrecision = TimestampPrecision.NS;
+                    long epochSec = value / 1000_000_000L;
+                    long nanoAdjustment = value % 1000_000_000L;
+                    return Timestamp.from(Instant.ofEpochSecond(epochSec, nanoAdjustment));
+                }
+            }
+            case UTC: {
+                String value = row.getString(colIndex);
+                if (value.lastIndexOf(":") > 19) {
+                    ZonedDateTime parse = ZonedDateTime.parse(value, rfc3339Parser);
+                    long nanoAdjustment;
+                    if (value.length() > 32) {
```

Comment:
is there a form way to pass precision instead of check the value length?

Reply:
There are many representations of UTC time, like `yyyy-MM-ddTHH:mm:ss.SSSSSSSSS+0x:00`,  `yyyy-MM-ddTHH:mm:ss.SSS+0x:00`, `yyyy-MM-ddTHH:mm:ss.SSSSSSSSS+0x00`, `yyyy-MM-ddTHH:mm:ss.SSS+0x00` and so on.  I need to deal with them separately. Do other languages have a more efficient way of doing this？

Reply:
we have the API taos_result_precision(TAOS_RES *res) to return precision. Can we get it from remote first?